### DISCUSSION TITLE
BusinessRule 'override' by user

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3381] [Questions]
We are writing several business rules that calculate fees/taxes etc. 
All of these rules could be overridden by 'supervisor' users.
We don't need to persist the 'override' property to the database.
How should we design this use case? 
Even if the 'supervisor' user enters a value different than what's calculated in the business rule, when the business object is persisted, it will save the value from the business rule(s).

Kind Regards

**Comment by rockfordlhotka** [comment]
There are at least a couple solutions to this sort of issue.

One is that the current user identity is changed to the overriding user for the duration of that user interacting with the app. Another is that an object graph (usually at the root) has a property that stores the overriding user identity.

I'm assuming that the override (and the overriding identity) needs to be logged, so it needs to either become the "current identity" or be reflected in the object graph during persistence.

I hear that you don't need to persist the override identity, though to me that seems a little odd - but business decisions are what they are.

Whether you persist the value or not, I'd suggest that the override identity (or maybe just the fact that an override occurred?) needs to be a property in the object graph.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka Could we use RuleSet in this case?
A RuleSet For Supervisor role vs regular employees?

This will be my first time using rulesets, this particular application gets it's roles and usernames from AD.
I can access the roles from ApplicationContext.User.Identity. I am thinking of creating a ruleset for each AD Role (AD roles used in my application).

Should I set the BusinessRule.RuleSet property in DataAccess code? I suppose I only need this in Create and Fetch methods? Since there couldn't be an Insert or Update without first calling Create or Fetch.

In cases where a user is a member of more than AD role and hence more than more RuleSet will apply.
how will that work within Csla?

Kind Regards.
